Apps/W1/Shopify/app/src/Companies/Codeunits/ShpfyCompByTaxId.Codeunit.al

```diff
@@ -0,0 +1,91 @@
+namespace Microsoft.Integration.Shopify;
+
+using Microsoft.Sales.Customer;
+
+/// <summary>
+/// Codeunit Shpfy Comp. By Tax Id (ID 30366) implements Interface Shpfy ICompany Mapping.
+/// </summary>
+codeunit 30366 "Shpfy Comp. By Tax Id" implements "Shpfy ICompany Mapping"
+{
+    Access = Internal;
+
+    internal procedure DoMapping(CompanyId: BigInteger; ShopCode: Code[20]; TemplateCode: Code[20]; AllowCreate: Boolean): Code[20]
+    var
+        ShopifyCompany: Record "Shpfy Company";
+    begin
+        ShopifyCompany.SetAutoCalcFields("Customer No.");
+        if ShopifyCompany.Get(CompanyId) then begin
+            if not IsNullGuid(ShopifyCompany."Customer SystemId") then begin
+                ShopifyCompany.CalcFields("Customer No.");
+                if ShopifyCompany."Customer No." = '' then begin
+                    Clear(ShopifyCompany."Customer SystemId");
+                    ShopifyCompany.Modify();
+                end else
+                    exit(ShopifyCompany."Customer No.");
+            end;
+            exit(CreateCompany(CompanyId, ShopCode, TemplateCode, AllowCreate));
+        end else
+            exit(CreateCompany(CompanyId, ShopCode, TemplateCode, AllowCreate));
+        exit('');
+    end;
+
+    internal procedure FindMapping(var ShopifyCompany: Record "Shpfy Company"; var TempShopifyCustomer: Record "Shpfy Customer" temporary): Boolean
+    var
+        Customer: Record Customer;
+        ShopifyCustomer: Record "Shpfy Customer";
+        Shop: Record "Shpfy Shop";
+        CompanyLocation: Record "Shpfy Company Location";
+        ShpfyTaxRegistrationIdMapping: Interface "Shpfy Tax Registration Id Mapping";
+    begin
+        if not IsNullGuid(ShopifyCompany."Customer SystemId") then
+            if Customer.GetBySystemId(ShopifyCompany."Customer SystemId") then
+                exit(true)
+            else begin
+                Clear(ShopifyCompany."Customer SystemId");
+                ShopifyCompany.Modify();
+            end;
+
+        if IsNullGuid(ShopifyCompany."Customer SystemId") then begin
```

Comment:
redundant begin--end

Also, here for readability, you could extract it to find by TaxReg or something